tlsn/tlsn-core/src/encoding/proof.rs

```diff
@@ -0,0 +1,158 @@
+        transcript_length: &TranscriptLength,
+        commitment: &EncodingCommitment,
+    ) -> Result<PartialTranscript, EncodingProofError> {
+        let seed: [u8; 32] = commitment
+            .seed
+            .clone()
+            .try_into()
+            .map_err(|_| EncodingProofError::InvalidSeed)?;
+
+        let encoder = new_encoder(seed);
+        let Self {
+            inclusion_proof,
+            openings,
+        } = self;
+        let (sent_len, recv_len) = (
+            transcript_length.sent as usize,
+            transcript_length.received as usize,
+        );
+
+        let mut indices = Vec::with_capacity(openings.len());
+        let mut leafs = Vec::with_capacity(openings.len());
+        let mut transcript = PartialTranscript::new(sent_len, recv_len);
+        let mut total_opened = 0u128;
+        for (id, Opening { seq, nonce }) in openings {
+            let opened_len = seq.data.len();
+
+            // Make sure the amount of data being proved is bounded.
+            total_opened += opened_len as u128;
+            if total_opened > MAX_TOTAL_COMMITTED_DATA as u128 {
+                return Err(EncodingProofError::ExceededMaxData);
+            }
+
+            // Make sure the opening length matches the ranges length.
+            if seq.data.len() != opened_len {
+                return Err(EncodingProofError::Malformed);
+            }
+
+            // Make sure the ranges are within the bounds of the transcript
+            let end = seq.idx.ranges.end().ok_or(EncodingProofError::EmptyRange)?;
+            let transcript_len = match seq.idx.direction {
+                Direction::Sent => sent_len,
+                Direction::Received => recv_len,
+            };
+
+            if end > transcript_len as usize {
+                return Err(EncodingProofError::OutOfBounds {
+                    input_end: end,
+                    transcript_length: transcript_len,
+                    direction: seq.idx.direction,
+                });
+            }
+
+            let expected_encoding = encoder.encode_subsequence(&seq.idx, &seq.data);
+            let expected_leaf = EncodingLeaf::new(expected_encoding, nonce);
+
+            // Compute the expected hash of the commitment to make sure it is
+            // present in the merkle tree.
+            indices.push(id);
+            leafs.push(expected_leaf);
+
+            // Union the authenticated subsequence into the transcript.
+            transcript.union_subsequence(&seq);
+        }
+
+        // Verify that the expected hashes are present in the merkle tree.
+        //
+        // This proves the Prover committed to the purported data prior to the encoder
+        // seed being revealed. Ergo, if the encodings are authentic then the purported
+        // data is authentic.
+        inclusion_proof
+            .verify(&commitment.root, &indices, &leafs)
+            .map_err(|_| EncodingProofError::MissingCommitment)?;
+
+        Ok(transcript)
```

Comment:
There is still no `sent` and `received` data in here, but I think this is correct at this place?! Should we add this as a comment for the function signature?

Reply:
Yeah, i think we can ignore it. I rechecked and it seems to be wrong.